### PR TITLE
Fix autonomous to wait for shooter wheel to be at speed 

### PR DESCRIPTION
**Pull request by imathno**

### CODE REVIEW

File: Competition2017/src/competition/subsystems/shoot_fuel/LeftShootFuelAndAgitateCommandGroup.java [hunk 1]
```diff
@@ -4,17 +4,21 @@
...
 import xbot.common.properties.XPropertyManager;
 
 public class LeftShootFuelAndAgitateCommandGroup extends CommandGroup {
     
     @Inject
-    public LeftShootFuelAndAgitateCommandGroup(XPropertyManager propManager, 
-            AgitatorsManagerSubsystem agitatorsManagerSubsystem,
-            LeftShootFuelCommandGroup shootLeft) {
+    public LeftShootFuelAndAgitateCommandGroup(AgitatorsManagerSubsystem agitatorsManagerSubsystem,
+                                               LeftShootFuelCommandGroup shootLeft,
```
Comment (WasabiFan) [inline Competition2017/src/competition/subsystems/shoot_fuel/LeftShootFuelAndAgitateCommandGroup.java]: Remove this indentation (and change your editor config to match)

File: Competition2017/src/competition/subsystems/shoot_fuel/LeftShootFuelAndAgitateCommandGroup.java [hunk 2]
```diff
@@ -4,17 +4,21 @@
...
+                                               ShooterWheelsManagerSubsystem shooterBeltsManagerSubsystem) {
         
-        IntakeAgitatorCommand intake = new IntakeAgitatorCommand(agitatorsManagerSubsystem.getLeftAgitator());
+        RunIntakeAgitatorIfWheelAtSpeedCommand intake = new RunIntakeAgitatorIfWheelAtSpeedCommand(
+                                                        agitatorsManagerSubsystem.getLeftAgitator(),
```
Comment (WasabiFan) [inline Competition2017/src/competition/subsystems/shoot_fuel/LeftShootFuelAndAgitateCommandGroup.java]: Same here

File: Competition2017/src/competition/subsystems/shoot_fuel/LeftShootFuelCommandGroup.java [hunk 3]
```diff
@@ -2,33 +2,27 @@
...
     @Inject
-    public LeftShootFuelCommandGroup(XPropertyManager propManager, 
-            AgitatorsManagerSubsystem agitatorsManagerSubsystem,
-            ShooterWheelsManagerSubsystem shooterWheelsManagerSubsystem,
-            ShooterBeltsManagerSubsystem shooterBeltsManagerSubsystem) {
-        IntakeAgitatorCommand intake = 
-                new IntakeAgitatorCommand(agitatorsManagerSubsystem.getLeftAgitator());
-        RunShooterWheelsForRangeCommand runWheel = 
+    public LeftShootFuelCommandGroup(ShooterWheelsManagerSubsystem shooterWheelsManagerSubsystem,
+                                     ShooterBeltsManagerSubsystem shooterBeltsManagerSubsystem) {
```
Comment (WasabiFan) [inline Competition2017/src/competition/subsystems/shoot_fuel/LeftShootFuelCommandGroup.java]: Again 😞

File: Competition2017/src/competition/subsystems/shoot_fuel/RightShootFuelCommandGroup.java [hunk 4]
```diff
@@ -2,33 +2,26 @@
...
                 new RunShooterWheelsForRangeCommand(
                         TypicalShootingPosition.FlushToBoiler, 
                         shooterWheelsManagerSubsystem.getRightShooter());
-        RunShooterBeltPowerCommand runBelt = 
-                new RunShooterBeltPowerCommand(shooterBeltsManagerSubsystem.getRightBelt());
-        
-        //this.addParallel(intake);
+        RunBeltIfWheelAtSpeedCommand runBelt =
+                new RunBeltIfWheelAtSpeedCommand(shooterBeltsManagerSubsystem.getRightBelt(),
+                                                    shooterWheelsManagerSubsystem.getRightShooter());
```
Comment (WasabiFan) [inline Competition2017/src/competition/subsystems/shoot_fuel/RightShootFuelCommandGroup.java]: Aaaargh

File: Competition2017/src/competition/subsystems/shoot_fuel/RightShootFuelAndAgitateCommandGroup.java [hunk 5]
```diff
@@ -4,17 +4,21 @@
...
 import xbot.common.properties.XPropertyManager;
 
 public class RightShootFuelAndAgitateCommandGroup extends CommandGroup {
     
     @Inject
-    public RightShootFuelAndAgitateCommandGroup(XPropertyManager propManager, 
-            AgitatorsManagerSubsystem agitatorsManagerSubsystem,
-            RightShootFuelCommandGroup shootRight) {
+    public RightShootFuelAndAgitateCommandGroup(AgitatorsManagerSubsystem agitatorsManagerSubsystem,
+                                                RightShootFuelCommandGroup shootRight,
```
Comment (WasabiFan) [inline Competition2017/src/competition/subsystems/shoot_fuel/RightShootFuelAndAgitateCommandGroup.java]: 💩

**Review by WasabiFan — DISMISSED** [review]

**Review by aschokking — APPROVED** [review]

File: Competition2017/src/competition/subsystems/shooter_wheel/ShooterWheelSubsystem.java [hunk 6]
```diff
@@ -98,7 +102,18 @@ public void setTargetSpeedForRange(double rangeInInches) {
     }
     
     public boolean isWheelAtSpeed() {
-        return super.getSpeed() >= super.getTargetSpeed() * wheelSpeedThresholdPercentage.get();
+        // Speed is in native units and can be negative
+        double currentSpeed = Math.abs(super.getSpeed());
+        if (currentSpeed <= 1) {
```
Comment (WasabiFan) [inline Competition2017/src/competition/subsystems/shooter_wheel/ShooterWheelSubsystem.java:107]: What is 1?

Comment (aschokking) [reply Competition2017/src/competition/subsystems/shooter_wheel/ShooterWheelSubsystem.java:107]: it was something arbitrarily close to zero, some smaller decimal would have worked too or we could have looked at the TargetSpeed which should have been exactly zero in this case probably.

Comment (WasabiFan) [reply Competition2017/src/competition/subsystems/shooter_wheel/ShooterWheelSubsystem.java:107]: But why is it needed? If our target speed is zero, then having a current speed of zero _is_ on-target. That state may indicate that the user chose an incorrect target, but I don't see a reason to special-case it as not being at the target speed when it clearly is for the domain of this system.

Comment (imathno) [reply Competition2017/src/competition/subsystems/shooter_wheel/ShooterWheelSubsystem.java:107]: The agitator was firing when we didn't have that if statement. It was a temporary fix but once we have time we should look back into the logic and fix it.

Comment (WasabiFan) [reply Competition2017/src/competition/subsystems/shooter_wheel/ShooterWheelSubsystem.java:107]: Ah, so the issue is that the target speed is zero when the agitators aren't requested to be moving, and that triggers the feeder. As I see it, the fix should be in the command that runs the feeders; this method is not solely used for that purpose.

Comment (WasabiFan) [reply Competition2017/src/competition/subsystems/shooter_wheel/ShooterWheelSubsystem.java:107]: I'll make this change and open a PR with the fix in a few minutes.

Comment (WasabiFan) [reply Competition2017/src/competition/subsystems/shooter_wheel/ShooterWheelSubsystem.java:107]: PR opened; thanks for the clarification!

**Pull request merged**